src/ipa_pc/mod.rs

```diff
@@ -1204,146 +1223,214 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
+                })?;
+
+            // y_i
+            let y_i = labeled_polynomial.polynomial().evaluate(*point);
+
+            // (X - x_i)
+            let x_polynomial = Polynomial::from_coefficients_slice(&[
+                (G::ScalarField::zero() - point),
+                G::ScalarField::one()
+            ]);
+
+            // z_i(X) = z(X) / (X - x_i)
+            let z_i_polynomial = &z_polynomial / &x_polynomial;
+
+            // z_i(x)
+            let z_i_value = z_i_polynomial.evaluate(x_point);
+
+            // (p_i(X) - y_i)
+            let polynomial =
+                &(labeled_polynomial.polynomial() - &Polynomial::from_coefficients_slice(&[y_i]));
+
+            // LC(p_1(X),p_2(X),...,p_m(X)) = SUM ( lamda^i * z_i(x) * (p_i(X) - y_i) )
+            lc_polynomial +=
+                (cur_challenge, &(polynomial * &Polynomial::from_coefficients_slice(&[z_i_value])));
+
+            if has_hiding {
+                lc_randomness.rand = lc_randomness.rand.iter().enumerate().map(|(index, &lc_rand_segment)| {
+                    if labeled_randomness.randomness().rand.len() > index {
+                        lc_rand_segment +
+                            labeled_randomness.randomness().rand[index] * cur_challenge * z_i_value
+                    } else {
+                        lc_rand_segment
+                    }
+                }).collect::<Vec<_>>();
+            }
+
+            // lambda^i
+            cur_challenge = cur_challenge * &lambda;
         }
 
-        // h(X) polynomial added to the set of polynomials for multi-poly single-point batching
-        let mut labeled_polynomials = labeled_polynomials;
+        // z(x)
+        let z_x_value = z_polynomial.evaluate(x_point);
+
+        // LC(p_1(X),p_2(X),...,p_m(X),h(X)) = SUM ( lamda^i * z_i(x) * (p_i(X) - y_i) ) - z(x) * h(X)
+        lc_polynomial += (-z_x_value, &h_polynomial);
+
+        let commit_time = start_timer!(|| format!("Commit to LC(p_1(X),p_2(X),...,p_m(X),h(X)) polynomial of degree {}", lc_polynomial.degree()));
+
+        // Commitment to the LC(p_1(X),p_2(X),...,p_m(X),h(X)) polynomial
+        let batch_comm = if lc_polynomial_len > key_len {
+            (0..lc_segments_count).into_iter().map(
+                |i| {
+                    Ok(Self::cm_commit(
+                        &ck.comm_key,
+                        &lc_polynomial.coeffs[i * key_len..core::cmp::min((i + 1) * key_len, lc_polynomial_len)],
+                        Some(ck.s),
+                        Some(lc_randomness.rand[i]),
+                    )?.into_affine())
+                }
+            ).collect::<Result<Vec<_>, _>>()?
+        } else {
+            vec![
+                Self::cm_commit(
+                    ck.comm_key.as_slice(),
+                    lc_polynomial.coeffs.as_slice(),
+                    Some(ck.s),
+                    Some(lc_randomness.rand[0]),
+                )?.into_affine()
+            ]
+        };
```

Comment:
This is overdoing it, and explains your above observation that the multi_point open does not need the commitments anymore: As the verifier, the commitment of the LC can be combined from the commitments of its components, avoiding another MSM.

Reply:
Done